src/gateway.rs

```diff
@@ -189,17 +193,20 @@ impl Gateway {
         }
     }
 
-    async fn handle_transmit_beacon(&mut self, logger: &Logger, beacon: Beacon) {
+    async fn handle_transmit_beacon(&mut self, logger: &Logger, beacon_req: BeaconReq) {
+        let BeaconReq { beacon, response } = beacon_req;
         let tx_power = if let Some(tx_power) = self.tx_power(logger) {
             tx_power
         } else {
+            response.send(Err(Error::custom("no tx power")), logger);
```

Comment:
these two custom errors are "unusual" but i'd still like to provide a better error response here, however i'm not completely sure where to create the module-level error for these: whether they belong in the `gateway.rs` module b/c that's where the error is being generated or in the `beaconer.rs` module b/c that's where they're handled and relevant.

Reply:
There's a RegionError in error.rs. This has a single enum entry for NoRegionParams. Add another entry there for `NoRegionTxPower`

I also believe we need to adjust `tx_power()` to return these RegionErrors and have the two callers of them log and fire up the error in the beacon case.

Reply:
hmm.. ok we could create a BeaconError enum to capture just the errors related to beacon _BUT_ have the region params error still be its own thing since `tx_power()` is used for both beacons and normal downlinks?